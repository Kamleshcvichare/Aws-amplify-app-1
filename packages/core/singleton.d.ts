export * from './lib-esm/Singleton';