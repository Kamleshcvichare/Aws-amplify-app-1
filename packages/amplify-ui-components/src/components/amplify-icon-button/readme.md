# amplify-icon-button

<!-- Auto Generated Below -->

## Properties

| Property          | Attribute           | Description | Type                                                                             | Default     |
| ----------------- | ------------------- | ----------- | -------------------------------------------------------------------------------- | ----------- |
| `autoShowTooltip` | `auto-show-tooltip` |             | `boolean`                                                                        | `undefined` |
| `name`            | `name`              |             | `"enter-vr" \| "exit-vr" \| "maximize" \| "minimize" \| "sound" \| "sound-mute"` | `undefined` |
| `tooltip`         | `tooltip`           |             | `string`                                                                         | `undefined` |

## Dependencies

### Used by

- [amplify-scene](../amplify-scene)

### Depends on

- [amplify-icon](../amplify-icon)

### Graph

```mermaid
graph TD;
  amplify-icon-button --> amplify-icon
  amplify-scene --> amplify-icon-button
  style amplify-icon-button fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
