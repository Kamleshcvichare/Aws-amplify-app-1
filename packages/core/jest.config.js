module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
