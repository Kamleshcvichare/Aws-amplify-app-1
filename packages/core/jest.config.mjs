import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
