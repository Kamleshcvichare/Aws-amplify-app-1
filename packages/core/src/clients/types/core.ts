/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
