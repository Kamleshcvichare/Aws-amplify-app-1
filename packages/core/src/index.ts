// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
This file maps top-level exports from `@aws-amplify/core`. These are intended to be potentially customer-facing exports.
*/
// Hub exports
export { Hub } from './Hub';
export { HubCapsule, HubCallback, HubPayload } from './Hub/types';

// Singleton exports
export {
	TokenProvider,
	AuthTokens,
	FetchAuthSessionOptions,
	AWSCredentialsAndIdentityIdProvider,
	AWSCredentialsAndIdentityId,
	Identity,
	OAuthConfig,
	CognitoUserPoolConfig,
} from './singleton/Auth/types';
export {
	AuthConfig,
	AuthUserPoolConfig,
	AuthUserPoolAndIdentityPoolConfig,
	APIConfig,
	CacheConfig,
	StorageAccessLevel,
	StorageConfig,
	GetCredentialsOptions,
	ResourcesConfig,
	LibraryOptions,
	AnalyticsConfig,
	GeoConfig,
} from './singleton/types';
export {
	Amplify,
	fetchAuthSession,
	AmplifyClass as AmplifyClassV6,
	clearCredentials,
} from './singleton';

// AWSClients exports
export {
	getCredentialsForIdentity,
	getId,
	GetCredentialsForIdentityInput,
	GetCredentialsForIdentityOutput,
} from './awsClients/cognitoIdentity';

// Amplify-wide constructs
export { UserProfile } from './types';

// Storage helpers
export {
	CookieStorage,
	defaultStorage,
	sessionStorage,
	sharedInMemoryStorage,
} from './storage';
export { KeyValueStorageInterface } from './types';

// Cache exports
export { Cache } from './Cache';

// Internationalization utilities
export { I18n } from './I18n';

// Logging utilities
export { ConsoleLogger } from './Logger';
