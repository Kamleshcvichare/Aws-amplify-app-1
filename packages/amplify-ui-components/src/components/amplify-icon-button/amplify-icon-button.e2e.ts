import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-icon-button e2e:', () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage();
    await page.setContent('<amplify-icon-button>FOO</amplify-icon-button>');
  });

  it(`renders with text 'FOO'`, async () => {
    const element = await page.find('amplify-icon-button');
    expect(element).not.toBeNull();

    const screenshot = await page.compareScreenshot('amplify-icon-button', { fullPage: true });
    expect(screenshot).toMatchScreenshot({ allowableMismatchedPixels: pixelThreshold });
  });
});
