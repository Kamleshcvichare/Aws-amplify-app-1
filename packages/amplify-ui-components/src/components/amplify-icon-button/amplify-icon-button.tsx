import { Component, Prop, h } from '@stencil/core';
import { actionButton } from './amplify-icon-button.style';
import { IconNameType } from '../amplify-icon/icons';

@Component({
  tag: 'amplify-icon-button',
  shadow: false,
})
export class AmplifySceneLoading {
  @Prop() name: IconNameType;
  @Prop() tooltip: string;
  @Prop() autoShowTooltip: boolean;

  render() {
    return (
      <button class={actionButton}>
        <amplify-icon name={this.name}></amplify-icon>
      </button>
    );
  }
}
