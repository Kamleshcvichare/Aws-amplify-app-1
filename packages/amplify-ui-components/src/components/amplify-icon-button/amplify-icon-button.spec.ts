import { newSpecPage } from '@stencil/core/testing';
import { AmplifyIconButton } from './amplify-icon-button';

describe('amplify-icon-button spec:', () => {
  describe('Render logic ->', () => {
    it('renders with no button text', async () => {
      const page = await newSpecPage({
        components: [AmplifyIconButton],
        html: `<amplify-icon-button></amplify-icon-button>`,
      });

      expect(page.root).toMatchSnapshot();
    });
    it('renders with button text FOO', async () => {
      const page = await newSpecPage({
        components: [AmplifyIconButton],
        html: `<amplify-icon-button>FOO</amplify-icon-button>`,
      });

      expect(page.root).toMatchSnapshot();
    });
    it('renders without Emotion CSS class when overrideStyle is true', async () => {
      const page = await newSpecPage({
        components: [AmplifyIconButton],
        html: `<amplify-icon-button override-style='true'>FOO</amplify-icon-button>`,
      });

      expect(page.root).toMatchSnapshot();
    });
    it('renders with danger class when button type is reset', async () => {
      const page = await newSpecPage({
        components: [AmplifyIconButton],
        html: `<amplify-icon-button type='reset'>FOO</amplify-icon-button>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
  describe('Component logic ->', () => {
    it('should have overrideStyle false by default', async () => {
      const button = new AmplifyIconButton();
      expect(button.overrideStyle).toBe(false);
    });
  });
});
