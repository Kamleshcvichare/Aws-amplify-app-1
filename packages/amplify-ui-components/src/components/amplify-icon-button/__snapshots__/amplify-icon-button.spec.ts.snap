// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-icon-button spec: Render logic -> renders enter-vr icon button correctly 1`] = `
<amplify-icon-button name="enter-vr">
  <button class="css-yqf6lz">
    <amplify-icon name="enter-vr"></amplify-icon>
  </button>
</amplify-icon-button>
`;

exports[`amplify-icon-button spec: Render logic -> renders exit-vr icon button correctly 1`] = `
<amplify-icon-button name="exit-vr">
  <button class="css-yqf6lz">
    <amplify-icon name="exit-vr"></amplify-icon>
  </button>
</amplify-icon-button>
`;

exports[`amplify-icon-button spec: Render logic -> renders maximize icon button correctly 1`] = `
<amplify-icon-button name="maximize">
  <button class="css-yqf6lz">
    <amplify-icon name="maximize"></amplify-icon>
  </button>
</amplify-icon-button>
`;

exports[`amplify-icon-button spec: Render logic -> renders minimize icon button correctly 1`] = `
<amplify-icon-button name="minimize">
  <button class="css-yqf6lz">
    <amplify-icon name="minimize"></amplify-icon>
  </button>
</amplify-icon-button>
`;

exports[`amplify-icon-button spec: Render logic -> renders sound icon button correctly 1`] = `
<amplify-icon-button name="sound">
  <button class="css-yqf6lz">
    <amplify-icon name="sound"></amplify-icon>
  </button>
</amplify-icon-button>
`;

exports[`amplify-icon-button spec: Render logic -> renders sound-mute icon button correctly 1`] = `
<amplify-icon-button name="sound-mute">
  <button class="css-yqf6lz">
    <amplify-icon name="sound-mute"></amplify-icon>
  </button>
</amplify-icon-button>
`;
